### DISCUSSION TITLE
Remote access "waiting for initialization"

### DISCUSSION THREAD

**Discussion by Sithuk** [zbm-dev/zfsbootmenu#597] [General]
I'm working on incorporating luks to the ubuntu zfsbootmenu install script. I've hit an issue with remote access that I hope someone can help with.

I've added agorgl's luks unlock script as a zfsbootmenu hook in early-setup.d. Link below to the script.
https://github.com/agorgl/zbm-luks-unlock

The unlock script works as expected. However, I encounter a problem when I add remote access and connect via ssh. ZFSBootMenu starts when I type "zfsbootmenu" in the ssh session, but I get a "Waiting for ZFSBootMenu initialization" screen. I suspect this is because the unlock script has been run as part of the boot sequence and it has not concluded as it is waiting for a password to be entered. One of the aims of remote access is to allow for ZFSBootMenu access when there is no local access. Is there a way to shutdown the script that is run by the hook or to override it to allow ZfsBootMenu to be started from ssh session? 

I have already tried running the luks-unlock script and it completes successfully from the ssh session. I just can't get into ZFSBootMenu to select the zpool and boot the system.

Can anyone offer any suggestions?

**Comment by zdykstra** [comment]
That script is blocking at a fairly early position in [zfsbootmenu-init](https://github.com/zbm-dev/zfsbootmenu/blob/master/zfsbootmenu/libexec/zfsbootmenu-init#L158-L160) . You can probably safely run `/libexec/zfsbootmenu-init` from your SSH session, but it's not really designed with that usage in mind so it could have unintended consequences.

**Comment by ahesford** [comment]
This should be resolved when #599 lands.

**Comment by Sithuk** [comment]
Fast work, thank you.

**Comment by Sithuk** [comment]
@ahesford : Do you have a roadmap for the next zfsbootmenu release that will have the fix? I would like to release the luks update for the ubuntu zfsbootmenu install script before Ubuntu 24.04 is released (April 25).

**Reply by zdykstra** [reply]
There's no set schedule for our releases. We've potentially collected enough changes and fixes to warrant one.

**Reply by Sithuk** [reply]
I've almost finished testing with 24.04 for the ubuntu zfsbootmenu install script. I'm currently pulling zfsbootmenu from master to get the remote access fix #599. I know you have said previously that pulling from master can introduce problems. Can you foresee any significant issues with using master until you release a new version of zfsbootmenu with #599 ?
